src/jit/gentree.h

```diff
@@ -574,11 +574,21 @@ struct GenTree
         return isContained() && IsCnsIntOrI() && !isUsedFromSpillTemp();
     }
 
+#if FEATURE_X87_DOUBLES
```

Comment:
Another orthogonal question is why do we need to represent float constant using a separate gentree node?  What optimizations are getting blocked due to current representation using GenTreeDblCon with type being TYP_FLOAT or TYP_DOUBLE?